src/providers/scripts/results.ts

```diff
@@ -57,6 +61,73 @@
                     const progressRing = document.createElement("vscode-progress-ring");
                     progressRing.id = "progress-ring";
                     container.appendChild(progressRing);
+
+                    // Record the start time of the query
+                    let startTime = Date.now();
+
+                    elapsedTime = 0;
+                    // Dynamically create the timer div if it doesn't exist
+                    if (!timerElement) {
+                        timerElement = document.createElement('div');
+                        timerElement.id = 'timer';
+                        timerElement.style.fontWeight = "300";
+                        timerElement.style.fontSize = "12px";
+                        timerElement.style.paddingTop = "0.5rem";
+                        document.body.appendChild(timerElement);
+                    }
+                    // Reset and show the timer content
+                    timerElement.textContent = 'Time elapsed:';
+                    timerElement.style.display = 'block';
+                    clearInterval(timer);
+                    timer = setInterval(() => {
+                        elapsedTime = Date.now() - startTime;
+                        let displayTime;
+
+                        // If less than 1000 ms, show full milliseconds
+                        if (elapsedTime < 1000) {
+                            displayTime = `${elapsedTime}ms`;
+                        } else {
+                            // After 1000 ms, display seconds with one decimal place
```

Comment:
The formatting function should be an exportable function and should reside in /utilities so that we can use it in both the Activity log and the elapsed time display after the query runs successfully.

Reply:
Neat! As discussed I'll go ahead and open an issue for this so we can take care of it in the react refactor.